src/cctag/filter/cvRecode.cpp

```diff
@@ -45,49 +45,42 @@ void cvRecodedCanny(
   int debug_info_level,
   const cctag::Parameters* params )
 {
-  CvMat srcCvMat = imgGraySrc;
-  CvMat *src = &srcCvMat;
-  
-  CvMat dstCvMat = imgCanny;
-  CvMat *dst = &dstCvMat;
-  
-  CvMat dxCvMat = imgDX;
-  CvMat *dx = &dxCvMat;
-  
-  CvMat dyCvMat = imgDY;
-  CvMat *dy = &dyCvMat;
-  
+  const cv::Mat& src = imgGraySrc;
+  cv::Mat& dst = imgCanny;
```

Comment:
if you don't mind, to make the code more readable, i'd remove dst, dx and dy and use directly imgCanny etc. in the rest of the code